scrapy/signalmanager.py

```diff
@@ -23,7 +28,16 @@ def connect(self, receiver, signal, **kwargs):
         :type signal: object
         """
         kwargs.setdefault('sender', self.sender)
-        return dispatcher.connect(receiver, signal, **kwargs)
+        if not func_accepts_kwargs(receiver):
+            warnings.warn("The use of handlers that don't accept "
+                          "**kwargs has been deprecated, plese refer "
+                          "to the Signals API documentation.",
+                          ScrapyDeprecationWarning, stacklevel=2)
+            self.signals[receiver.__repr__()] = \
```

Comment:
@jdemaeyer I read [here](http://stackoverflow.com/questions/11162201/is-it-always-faster-to-use-string-as-key-in-a-dict) that it's best to use string keys in a Python dictionary, so decided to use the string representation of the callable as key instead of a callable instance.